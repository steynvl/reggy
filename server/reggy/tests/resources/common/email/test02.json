{
  "type": "Email address",
  "information": {
    "username": "Basic characters only [a-z0-9._-]",
    "domainName": "Allow any domain name",
    "mailtoPrefix": "Optional prefix",
    "specificUserNamesOnly": "",
    "domainOnSpecificTld": "",
    "anySubDomainOnSpecificDomain": "",
    "specificDomainsOnly": ""
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}