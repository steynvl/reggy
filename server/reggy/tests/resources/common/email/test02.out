Java
Pattern regex = Pattern.compile("(?:mailto:)?+[\\d_a-z-]++(?:\\.[\\d_a-z-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+");
(?:mailto:)?+[\\d_a-z-]++(?:\\.[\\d_a-z-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+
Perl
my $regex = /(?:mailto:)?+[\d_a-z-]++(?:\.[\d_a-z-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+/;
(?:mailto:)?+[\d_a-z-]++(?:\.[\d_a-z-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+
POSIX
(?:mailto:)?+[\d_a-z-]++(?:\.[\d_a-z-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+
(?:mailto:)?+[\d_a-z-]++(?:\.[\d_a-z-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+
Python
regex = re.compile(r'(?:mailto:)?+[\d_a-z-]+(?:\.[\d_a-z-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}')
(?:mailto:)?+[\d_a-z-]+(?:\.[\d_a-z-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}
JavaScript
const regex = /(?:mailto:)?+[\d_a-z-]+(?:\.[\d_a-z-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}/;
(?:mailto:)?+[\d_a-z-]+(?:\.[\d_a-z-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}
PHP
$regex = '/(?:mailto:)?+[\d_a-z-]++(?:\.[\d_a-z-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+/';
(?:mailto:)?+[\d_a-z-]++(?:\.[\d_a-z-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+
Golang
regex, _ := regexp.Compile("(?:mailto:)?+[\\d_a-z-]+(?:\\.[\\d_a-z-]+)*@(?:[\\da-z-]+\\.)+[a-z]{2,63}")
(?:mailto:)?+[\\d_a-z-]+(?:\\.[\\d_a-z-]+)*@(?:[\\da-z-]+\\.)+[a-z]{2,63}
Rust
let re = Regex::new(r"(?:mailto:)?+[\d_a-z-]+(?:\.[\d_a-z-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}").unwrap();
(?:mailto:)?+[\d_a-z-]+(?:\.[\d_a-z-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}
C#
var re = new Regex(@"(?:mailto:)?+[\d_a-z-]+(?:\.[\d_a-z-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}");
(?:mailto:)?+[\d_a-z-]+(?:\.[\d_a-z-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}
Scala
val re = "(?:mailto:)?+[\\d_a-z-]++(?:\\.[\\d_a-z-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+".r
(?:mailto:)?+[\\d_a-z-]++(?:\\.[\\d_a-z-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+
Kotlin
val regex = Regex("(?:mailto:)?+[\\d_a-z-]++(?:\\.[\\d_a-z-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+")
(?:mailto:)?+[\\d_a-z-]++(?:\\.[\\d_a-z-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+